---
title: "Class 13: Pathway Analysis from RNA-Seq Results"
author: "Angela Bartolo PID: A15932451"
format: pdf
---

## Section 1. Differential Expression Analysis

```{r}
library(DESeq2)
```

```{r}
colData <- read.csv("GSE37704_metadata.csv")
head(colData)

countData <- read.csv("GSE37704_featurecounts.csv", row.names =1)
head(countData)
```

To match the countData with the colData, the length column in countData must be removed. 

> Q. Complete the code below to remove the troublesome first column from countData

```{r}
# Note we need to remove the odd first $length col
countData <- as.matrix(countData[,colData$id])
head(countData)
```

We must remove the zeros in the entries.

> Q. Complete the code below to filter countData to exclude genes (i.e. rows) where we have 0 read count across all samples (i.e. columns).

```{r}
zeros <- rowSums(countData) > 0 
countData = countData[zeros, ]
head(countData)
```

```{r}
nrow(countData)
```

## 3. Setup and run DESeq2

```{r}
#/ message:false
library(DESeq2)
```

```{r}
dds = DESeqDataSetFromMatrix(countData=countData,
                             colData=colData,
                             design=~condition)
dds = DESeq(dds)
```

```{r}
res <- results(dds)
head(res)
```


## 4. Annotation results

I need to add annotation to my results including gene symbols and entrezids et. For this I will use the Annotationdbi package

```{r}
library(AnnotationDbi)
library("org.Hs.eg.db")

columns(org.Hs.eg.db)
```

```{r}
res$symbol = mapIds(org.Hs.eg.db,
                    keys=row.names(res), 
                    keytype="ENSEMBL",
                    column="SYMBOL",
                    multiVals="first")

res$entrez = mapIds(org.Hs.eg.db,
                    keys=row.names(res),
                    keytype="ENSEMBL",
                    column="ENTREZID",
                    multiVals="first")

```

```{r}
head(res)
```
```{r}
res = res[order(res$pvalue),]
write.csv(res, file ="deseq_results.csv")
```

```{r}
plot( res$log2FoldChange, -log(res$padj) )
```

```{r}
#/ message:false
library(pathview)
library(gage)
library(gageData)
```

Focus in on signaling pathways
```{r}
data(kegg.sets.hs)
data(sigmet.idx.hs)

# Focus on signaling and metabolic pathways only
kegg.sets.hs = kegg.sets.hs[sigmet.idx.hs]

# Examine the first 3 pathways
head(kegg.sets.hs, 3)
```
```{r}
foldchanges = res$log2FoldChange
names(foldchanges) = res$entrez
head(foldchanges)
```

Run gauge
```{r}
#Get the results
keggres = gage(foldchanges, gsets=kegg.sets.hs)
```

Look at the first few down (less) pathways
```{r}
attributes(keggres)
head(keggres$less)
```

```{r}
pathview(gene.data = foldchanges, pathway.id="hsa04110")
```

![](hsa04110.pathview.png) 


